Zeeguu Reader/Zeeguu Reader/LoginTableViewController.swift

```diff
@@ -89,11 +89,15 @@ class LoginTableViewController: ZGTableViewController {
 		
 		if let em = email, pw = password {
 			ZeeguuAPI.sharedAPI().loginWithEmail(em, password: pw) { (success) -> Void in
-				if (success) {
-					self.dismissViewControllerAnimated(true, completion: nil)
-				} else {
-					Utils.showOKAlertWithTitle("LOGIN_FAILED".localized, message: "LOGIN_FAILED_MESSAGE".localized, okAction: nil)
-				}
+				dispatch_async(dispatch_get_main_queue(), {
```

Comment:
No. The block that you pass the ZeeguuAPI endpoint is executed in the background. This `dispatch_async` function ensures that the block of code you pass to it is executed on the main thread (as given by `dispatch_get_main_queue()`). I send it to the main thread, because there is some GUI stuff happening, which may or may not crash/give warnings if executed in the background.

See also [this](http://stackoverflow.com/questions/16283652/understanding-dispatch-async/17453020#17453020) answer, which explains a bit about the GCD queues, or [this](https://en.wikipedia.org/wiki/Grand_Central_Dispatch) link, which explains what Grand Central Dispatch is.